### PR TITLE
Fixing activityContext propagation

### LINKED ISSUE #1445: HttpWebRequest instrumentation must propagate Current context
Similar to this: https://github.com/open-telemetry/opentelemetry-dotnet/pull/1431.

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #1445.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs [hunk 1]
```diff
@@ -214,7 +212,9 @@ private static void ProcessRequest(HttpWebRequest request)
             }
 
             var activity = WebRequestActivitySource.StartActivity(ActivityName, ActivityKind.Client);
+            var activityContext = Activity.Current?.Context ?? default;
 
+            InstrumentRequest(request, activityContext);
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:217]: AddRequestTagsAndInstrumentRequest() down the line will call InstrumentRequest Again, so need to chaneg that too

Comment (eddynaka) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:217]: I removed from that part. So, now we just have one call

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestTests.Basic.netfx.cs [hunk 2]
```diff
@@ -95,6 +95,43 @@ public async Task HttpWebRequestInstrumentationInjectsHeadersAsync()
...
+
+            var parent = new Activity("parent")
+                .SetIdFormat(ActivityIdFormat.W3C)
+                .Start();
+            parent.TraceStateString = "k1=v1,k2=v2";
+            parent.ActivityTraceFlags = ActivityTraceFlags.None;
+
+            using var response = await request.GetResponseAsync();
+
+            Assert.Empty(activityProcessor.Invocations);
```
Comment (cijothomas) [inline test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestTests.Basic.netfx.cs:119]: how is this empty?

Comment (cijothomas) [reply test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestTests.Basic.netfx.cs:119]: its best to add comment like:
By default parentbasedsampler is used.
In this case, the parent is the manually created parentactivity, which will have TraceFlags as None.
This causes child to be *not* created.

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestTests.Basic.netfx.cs [hunk 3]
```diff
@@ -95,6 +95,43 @@ public async Task HttpWebRequestInstrumentationInjectsHeadersAsync()
...
+            var activity = Activity.Current;
+
+            Assert.Equal(parent.TraceId, activity.Context.TraceId);
```
Comment (cijothomas) [inline test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestTests.Basic.netfx.cs:122]: i think what you want to do is to register a customPropagator, and inside that validate that ActivityContext is the one from parent

Comment (eddynaka) [reply test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestTests.Basic.netfx.cs]: just updated. i will might have to change after this or after the other pr is merged.

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestTests.Basic.netfx.cs (hunk 2 above)
Comment (eddynaka) [reply test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestTests.Basic.netfx.cs:128]: added. thanks!

**Review by cijothomas — APPROVED** [review]

**Pull request merged**